### PR TITLE
Support Apple app bundle signing for already archived payloads

### LINKED ISSUE #7815: XHarness Helix SDK doesn't support Apple devices when zipped app provided
### Problem

When we prepare Helix payloads in Helix SDK ([CreateXHarnessAppleWorkItems.cs](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Helix/Sdk/CreateXHarnessAppleWorkItems.cs)) and we are targeting iOS or tvOS devices, we inject the provisioning profile in every app bundle's folder.

The SDK also supports providing an already packaged .zip with the app bundle in which it only adds the shell scripts before sending it.

The code will probably crash if it tries to copy the provisioning profile in the zip as it doesn't know about this feature.

### Goal

Make the [ProvisioningProfileProvider](https://github.com/dotnet/arcade/blob/761d2e6545a31d2ebf9cb60443ed2a9c2f8268b5/src/Microsoft.DotNet.Helix/Sdk/ProvisioningProfileProvider.cs#L85) also recognize that the incoming app bundle is already zipped and inject the provisioning profile in there too.

### PR DESCRIPTION
**Pull request by premun**
When user provides an already archived application (or more), we need to inject the provisioning profile in each of the app bundles inside of the archive. This was not supported.

If there are no app bundles (might be built as part of the Helix work item), we only store the profile in the root of the payload.

Resolves #7815

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/ProvisioningProfileProvider.cs [hunk 1]
```diff
@@ -36,100 +50,128 @@ public class ProvisioningProfileProvider : IProvisioningProfileProvider
...
+                else
                 {
-                    var platform = pair.Key;
-                    var targetName = pair.Value;
-
-                    if (!bundleTargets.Contains(targetName))
-                    {
-                        continue;
-                    }
+                    rootLevelAppBundles.Add(appBundleName);
```
Comment (jakubstilec) [inline src/Microsoft.DotNet.Helix/Sdk/ProvisioningProfileProvider.cs:141]: Here goes everything else that doesn't end with  "/" + ProfileFileName. Is it correct? Shouldn't we add some logging in a case there are unexpected name?

File: src/Microsoft.DotNet.Helix/Sdk/CreateXHarnessAppleWorkItems.cs [hunk 2]
```diff
@@ -107,6 +107,17 @@ public override void ConfigureServices(IServiceCollection collection)
                 Log.LogError($"App bundle not found in {appFolderPath}");
                 return null;
             }
+
+            // If we are re-using one .zip for multiple work items, we need to copy it to a new location
+            // because we will be changing the contents (we assume we don't mind otherwise)
+            if (isAlreadyArchived && appBundleItem.TryGetMetadata(MetadataNames.AppBundlePath, out string metadata) && !string.IsNullOrEmpty(metadata))
+            {
+                string appFolderDirectory = fileSystem.GetDirectoryName(appFolderPath);
+                string fileName = $"xharness-payload-{workItemName.ToLowerInvariant()}.zip";
```
[the remark this answers is not in the document]
Comment (premun) [reply src/Microsoft.DotNet.Helix/Sdk/CreateXHarnessAppleWorkItems.cs:116]: They must be unique

File: src/Microsoft.DotNet.Helix/Sdk/ProvisioningProfileProvider.cs (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.Helix/Sdk/ProvisioningProfileProvider.cs:141]: Here go all files whose path looks like this `[appBundleName].app/**/*`. Ultimately, we want to end up with a record for every `appBundleName` in root minus the ones that have the profile already (which is the other HashSet).

So there's nothing like "unexpected name" really. Or do you mean something else?

**Review by jakubstilec — APPROVED** [review]

**Pull request merged**